### ISSUE TITLE
Documentation error in spherical Bessel, Neumann, modified spherical Bessel, Neumann

### ISSUE THREAD

**Issue by amdj** [scipy/scipy#7787]
The documentation of 
- https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.spherical_jn.html
- https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.spherical_yn.html
- https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.spherical_in.html
- https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.spherical_kn.html

says (for example for the spherical_jn)
```
    .. math::
        j_n' = j_{n-1} - \frac{n + 1}{2} j_n.
        j_0' = -j_1

```
, which should be

```
    .. math::
        j_n' = j_{n-1} - \frac{n + 1}{z} j_n.
        j_0' = -j_1

```
as is correctly stated in the source at
[http://dlmf.nist.gov/10.51#E2](http://dlmf.nist.gov/10.51#E2)

**Comment by ev-br** [comment]
Thanks for the report! Would you be interested in submitting a PR witg the fix?

**Comment by amdj** [comment]
Yes, no problem.

**Comment by amdj** [comment]
I found that is already solved by annesylvie, commit id a78ff18d. This however never got into the version releases. It was fixed, but never merged into a branch corresponding to a version. What can I do beyond this point?

Thanks in advance,
Anne

**Comment by ev-br** [comment]
Indeed, https://scipy.github.io/devdocs/generated/scipy.special.spherical_jn.html looks correct. So this will be available in the next version of scipy, when it's released.

However 
https://scipy.github.io/devdocs/generated/scipy.special.spherical_yn.html
does not look right still.
